### PR TITLE
REV-150-agent-profile-rate-and-review

### PR DESCRIPTION
**Pull request by IbrahimAldereni-socuimtech**
### Jira URL

https://optimumpartners.atlassian.net/browse/REV-150

### Description

- create add, update, get user reviews
- figma link: https://www.figma.com/file/IKofGteYvhvMoXoafbBpgi/reverifi?node-id=1692%3A6631

### Dependencies in Other PRs,Repo

- backend PR: https://github.com/abukhadraop-or/reverifi-api/pull/36

Screenshot from 2022-02-23 17-20-56
Screenshot from 2022-02-23 17-21-04

### CODE REVIEW

File: src/components/agent-details/AgentRating/agent-rating-schema.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import * as yup from 'yup';
+
+const agentRatingSchema = yup
+  .object({
+    review: yup
+      .string()
+      .required('Please fill out this field')
```
Comment (AhmadAlShwaiki) [inline src/components/agent-details/AgentRating/agent-rating-schema.js:7]:
```suggestion
      .required('This field can not be empty')
```

Comment (IbrahimAldereni-socuimtech) [reply src/components/agent-details/AgentRating/agent-rating-schema.js:7]: that's what Duah wants

File: src/components/agent-details/AgentRating/agent-rating.style.js [hunk 2]
```diff
@@ -23,26 +26,34 @@ export const RatingForm = styled.form`
...
-  font-family: inherit;
-  font-size: 1rem;
-  box-shadow: 0 0.06rem 0.38rem ${colors.mercury};
+  > button {
+    font-weight: 600;
+    font-size: 0.9rem;
+    align-self: flex-start;
+  }
+
+  textarea {
```
[the remark this answers is not in the document]
Comment (IbrahimAldereni-socuimtech) [reply src/components/agent-details/AgentRating/agent-rating.style.js:43]: I am using shared component

File: src/components/shared/LikeButton/index.jsx [hunk 3]
```diff
@@ -1,22 +1,54 @@
...
+import React, { useEffect, useState } from 'react';
 import { LikeButtonWrapper } from './like-button.styles';
 
 /**
  * Like Button component
  *
  * @return {JSX.Element}
  */
-function LikeButton() {
+function LikeButton({ grayIcon, active }) {
```
Comment (mohdabed123) [inline src/components/shared/LikeButton/index.jsx:14]:
```suggestion
function LikeButton({ isLiked, active }) {
```

File: src/components/shared/FormTextArea/form-text-area.styles.js [hunk 4]
```diff
@@ -3,7 +3,8 @@ import colors from 'styles/colors';
 
 export const Labels = styled.div`
   display: flex;
-  justify-content: space-between;
+  justify-content: ${({ isLabeled }) =>
+    isLabeled ? 'space-between' : 'flex-end'};
```
Comment (mohdabed123) [inline src/components/shared/FormTextArea/form-text-area.styles.js:7]:
```suggestion
    hasLabel ? 'space-between' : 'flex-end'};
```

**Review by mohdabed123 — CHANGES_REQUESTED** [review]

File: src/components/agent-details/AgentRating/agent-rating-schema.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mohdabed123) [reply src/components/agent-details/AgentRating/agent-rating-schema.js:7]: you can use either this one or the one ahmad recommended as you see fit

[the remark this answers is not in the document]
Comment (IbrahimAldereni-socuimtech) [reply src/components/agent-details/AgentRating/agent-rating-schema.js:7]: that's the recommandation of Duha

File: src/components/shared/LikeButton/index.jsx (hunk 3 above)
Comment (IbrahimAldereni-socuimtech) [reply src/components/shared/LikeButton/index.jsx:14]: thats for styling, sometimes we use different icons colors, not for functinality

**Review by AhmadAlShwaiki — APPROVED** [review]

**Pull request merged**